lib/components/user/saved-trip-screen.js

```diff
@@ -102,12 +103,35 @@ class SavedTripScreen extends Component {
 
   _handleSaveNewTrip = async () => {
     await this._updateMonitoredTrip()
-    this._goToTripPlanner()
+
+    // For new trips, if a trip with the same ui_activeSearch value in queryParams
+    // appears in the monitored trips redux state,
+    // then we know that save was successful and we can proceed.
+    const { activeSearchId } = this.props
+
+    const savedTrip = this.props.monitoredTrips.find(trip => {
+      const tripQueryParams = qs.parse(trip.queryParams.split('?')[1])
+      return tripQueryParams.ui_activeSearch === activeSearchId
+    })
```

Comment:
This approach seems flimsy. I would prefer that this logic be contained within the action -- that would remove all of this inference that is being applied here. I think the routing logic for these handlers (goToTripPlanner, goToSavedTrips, etc.) should also be contained within those actions.